### PR TITLE
Addon-options back compat

### PR DESCRIPTION
**Pull request by shilman**
Issue: #5665 #5671

## What I did
- Add back compatibility for options
- Add migration instructions
- Add deprecation notice in README
- Update official docs

## How to test

Update official storybook to use deprecated options like `goFullScreen` or `addonPanelInRight`

### CODE REVIEW

File: MIGRATION.md [hunk 1]
```diff
@@ -88,6 +89,52 @@ addParameters({
...
+And here's its new counterpart:
+
+```js
+import { create } from '@storybook/theming';
+addParameters({
+  options: {
+    theme: create({
+      brandTitle: 'Storybook',
+      brandUrl: 'https://storybook.js.org'
+    }, 'normal'),
```
Comment (tmeasday) [inline MIGRATION.md]: Is this right (`'normal'`?)

File: MIGRATION.md [hunk 2]
```diff
@@ -88,6 +89,52 @@ addParameters({
...
+| Old               | New              |
+| ----------------- | ---------------- |
+| name              | theme.brandTitle |
+| url               | theme.brandUrl   |
+| goFullScreen      | isFullscreen     |
+| showStoriesPanel  | showNav          |
+| showAddonPanel    | showPanel        |
+| showSearchBox     |                  |
+| addonPanelInRight | panelPosition    |
+| selectedPanel     |                  |
```
Comment (tmeasday) [inline MIGRATION.md]: Are the ones that are missing here no longer supported?

**Review by tmeasday — COMMENTED** [review]
LGTM a couple of questions

File: MIGRATION.md [hunk 3]
```diff
@@ -115,7 +115,7 @@ addParameters({
     theme: create({
       brandTitle: 'Storybook',
       brandUrl: 'https://storybook.js.org'
-    }, 'normal'),
+    }, 'light'),
```
Comment (tmeasday) [inline MIGRATION.md]: The second argument to `create` is an object: https://github.com/storybooks/storybook/blob/next/lib/theming/src/create.ts#L113

**Review by tmeasday — DISMISSED** [review]

File: docs/src/pages/configurations/options-parameter/index.md [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+The options-addon accepts story parameters on the `options` key:
+
+```js
+import { storiesOf } from '@storybook/react';
+import MyComponent from './my-component';
+
+storiesOf('Addons|Custom options', module)
+  // If you want to set the option for all stories in of this kind
+  .addParameters({ options: { addonPanelInRight: true } })
```
Comment (tmeasday) [inline docs/src/pages/configurations/options-parameter/index.md]: `panelPosition`?

**Comment by shilman** [comment]
Waiting on #5773 since this touches the same code

File: MIGRATION.md (hunk 2 above)
Comment (shilman) [reply MIGRATION.md]: Updated

**Pull request merged**